dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/JsonPathSkill.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Linq;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI;
+
+public class JsonPathSkill
+{
+    /// <summary>
+    /// <see cref="ContextVariables"/> parameter names.
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// JSON path.
+        /// </summary>
+        public const string JsonPath = "jsonpath";
+    }
+
+    /// <summary>
+    /// Retrieve one or more values from a JSON string using a JsonPath query.
+    /// </summary>
+    [SKFunction("Retrieve a single value of a property from a JSON string.")]
+    [SKFunctionInput(Description = "JSON string")]
+    [SKFunctionContextParameter(Name = "JsonPath", Description = "JSON path.")]
+    public string GetJsonPropertyValue(string json, SKContext context)
+    {
+        if (string.IsNullOrWhiteSpace(json))
+        {
+            context.Fail($"Missing input JSON.");
+            return string.Empty;
+        }
+
+        if (!context.Variables.Get(Parameters.JsonPath, out string jsonPath))
+        {
+            context.Fail($"Missing variable {Parameters.JsonPath}.");
+            return string.Empty;
+        }
+
+        JObject jsonObject = JObject.Parse(json);
+
+        JToken[] tokens = jsonObject.SelectTokens(jsonPath).ToArray();
+
+        if (!tokens.Any())
+        {
+            // If there were no results, return an empty string.
+            return string.Empty;
+        }
+        else if (tokens.Length == 1 &&
+            tokens.First().GetType() == typeof(JValue))
+        {
+            // If there was a single result that is a simple value, return just that value.
+            return tokens.First().Value<string>() ?? string.Empty;
```

Comment:
As the caller of this function (or the next one in the line), how would I know if the value I'm getting is going to be a string value or structured json? Tryparse the result; then if that fails, and string is not empty, assume it's the string value?

No great ideas here...
 - always return json (even for sinlgle values)
 - always return only the first value, ignore complex types (perhaps with more SKFunctions supporting other non-value types, FindFirst, FindAll)
 - set another SKContext parameter that describes the returned value type (JArray, JObject... or string)

Reply:
My original draft had two SK Functions - one intended for singular values and one intended for complex/array values. I'll split the SK functions this way again which should take care of your concern here. We can always collapse them in the future after feedback from users.